org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/popup/AnimationUtil.java

```diff
@@ -0,0 +1,163 @@
+    return new FadeJob(shell, 2 * FADE_IN_INCREMENT, FADE_RESCHEDULE_DELAY, listener);
+  }
+
+  public static FadeJob fadeIn(Shell shell, IFadeListener listener) {
+    return new FadeJob(shell, FADE_IN_INCREMENT, FADE_RESCHEDULE_DELAY, listener);
+  }
+
+  public static FadeJob fadeOut(Shell shell, IFadeListener listener) {
+    return new FadeJob(shell, FADE_OUT_INCREMENT, FADE_RESCHEDULE_DELAY, listener);
+  }
+
+  public static class FadeJob extends Job {
+
+    private final Shell shell;
+
+    private final int increment;
+
+    private volatile boolean stopped;
+
+    private volatile int currentAlpha;
+
+    private final long delay;
+
+    private final IFadeListener fadeListener;
+
+    public FadeJob(Shell shell, int increment, long delay, IFadeListener fadeListener) {
+      super("Fading");
+      if (increment < -255 || increment == 0 || increment > 255) {
+        throw new IllegalArgumentException("-255 <= increment <= 255 && increment != 0"); //$NON-NLS-1$
+      }
+      if (delay < 1) {
+        throw new IllegalArgumentException("delay must be > 0"); //$NON-NLS-1$
+      }
+      this.currentAlpha = shell.getAlpha();
+      this.shell = shell;
+      this.increment = increment;
+      this.delay = delay;
+      this.fadeListener = fadeListener;
+
+      setSystem(true);
+      schedule(delay);
+    }
+
+    @Override
+    protected void canceling() {
+      stopped = true;
+    }
+
+    private void reschedule() {
+      if (stopped) {
+        return;
+      }
+      schedule(delay);
+    }
+
+    public void cancelAndWait(final boolean setAlpha) {
+      if (stopped) {
+        return;
+      }
+      cancel();
+      Display.getDefault().syncExec(new Runnable() {
+        public void run() {
+          if (setAlpha) {
+            shell.setAlpha(getLastAlpha());
+          }
+        }
+      });
+    }
+
+    @Override
+    protected IStatus run(IProgressMonitor monitor) {
+      if (stopped) {
+        return Status.OK_STATUS;
+      }
+
+      currentAlpha += increment;
+      if (currentAlpha <= 0) {
+        currentAlpha = 0;
+      } else if (currentAlpha >= 255) {
+        currentAlpha = 255;
+      }
+
+      Display.getDefault().syncExec(new Runnable() {
+        public void run() {
+          if (stopped) {
+            return;
+          }
+
+          if (shell.isDisposed()) {
+            stopped = true;
+            return;
+          }
+
+          shell.setAlpha(currentAlpha);
+
+          if (fadeListener != null) {
+            fadeListener.faded(shell, currentAlpha);
+          }
+        }
+      });
+
+      if (currentAlpha == 0 || currentAlpha == 255) {
+        stopped = true;
+      }
+
+      reschedule();
+      return Status.OK_STATUS;
+    }
+
+    private int getLastAlpha() {
+      return (increment < 0) ? 0 : 255;
+    }
+
+  }
+
+  public static interface IFadeListener {
+
+    public void faded(Shell shell, int alpha);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) "public" is redundant in this context. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2333)